acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -192,6 +211,32 @@ impl BlackBoxFuncCall {
                 inputs.extend(hashed_message.iter().copied());
                 inputs
             }
+            BlackBoxFuncCall::VerifyProof {
+                key,
+                proof,
+                public_inputs,
+                key_hash,
+                input_aggregation_object,
+                ..
+            } => {
+                let mut inputs = Vec::with_capacity(
+                    key.len()
+                        + proof.len()
+                        + public_inputs.len()
+                        + 1
+                        + input_aggregation_object.len(),
+                );
+                inputs.extend(key.iter().copied());
+                inputs.extend(proof.iter().copied());
+                inputs.extend(public_inputs.iter().copied());
+                inputs.push(*key_hash);
+                // If we do not have an input aggregation object assigned
+                // do not return it as part of the input vector as to not trigger
+                if !input_aggregation_object.iter().any(|v| v.witness == Witness(0)) {
```

Comment:
Witness(0) seems to be a barretenberg specific idiom?

Reply:
So I was debating how to go about this here. I could add a flag that determines whether or not we should include the `input_aggregation_object` in our inputs vector instead of this check. If the `input_aggregation_object` is all zeros we will get a missing assignment error when we call `solve` in `blackbox.rs`. Setting the `input_aggregation_object` as all zeros is specific to our bberg recursion circuit, but either way we are going to have to some sort of indicator whether to have an input aggregation object assigned. I was thinking a flag is similar to requiring anyone who is integrating the VerifyProof blackbox func to have the `input_aggregation_object` of their first recursive verification be all zeros. However, a flag is more explicit and I can switch to this

Reply:
We don't have anything in acvm right now that states that Witness(0) is special

Can we have `input_aggregation_object` be optional then explain what it means for it to be None?

Reply:
Yes I can switch to that instead

Reply:
I switched to an Option. It is now entirely up to a backend to determine how they will handle the first recursive aggregation